wallet/psbt.go

```diff
@@ -390,6 +390,39 @@ func (w *Wallet) FinalizePsbt(keyScope *waddrmgr.KeyScope, account uint32,
 	return nil
 }
 
+// PsbtPrevOutputFetcher returns a txscript.PrevOutFetcher built from the UTXO
+// information in a PSBT packet.
+func PsbtPrevOutputFetcher(packet *psbt.Packet) *txscript.MultiPrevOutFetcher {
```

Comment:
👍 

I think we can keep the PSBT scope minimal here so we'll push off the interpretation of all the new fields to another PR.